### PR TITLE
NIFI-2971 moved properties loading into its own bootstrap classloader

### PR DESCRIPTION
**Pull request by joewitt**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/main/java/org/apache/nifi/NiFi.java [hunk 1]
```diff
@@ -55,6 +59,25 @@
...
+        try {
+            Files.list(Paths.get("lib/bootstrap")).forEach(p -> {
+                try {
+                    urls.add(p.toUri().toURL());
+                } catch (final MalformedURLException mef) {
+                    logger.warn("Unable to load " + p.getFileName() + " due to " + mef, mef);
+                }
+            });
+        } catch (final IOException ex) {
+            logger.error("Unable to properly initialize bootstrap classloader due to " + ex, ex);
```
Comment (olegz) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/main/java/org/apache/nifi/NiFi.java]: Does this mean we can/should continue?

Comment (joewitt) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-runtime/src/main/java/org/apache/nifi/NiFi.java]: Will fix that and will make the following changes which ensure RPM output aligns to tar output in terms of lib/ and lib/bootstrap contents


---
 nifi-assembly/pom.xml | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/nifi-assembly/pom.xml b/nifi-assembly/pom.xml
index d9f4f99..5eef82f 100755
--- a/nifi-assembly/pom.xml
+++ b/nifi-assembly/pom.xml
@@ -538,8 +538,8 @@ language governing permissions and limitations under the License. -->
                                                 <excludes>
                                                     <exclude>org.bouncycastle:bcprov-jdk15on</exclude>
                                                     <exclude>org.apache.commons:commons-lang3</exclude>
-                                                    <exclude>org.slf4j:jul-to-slf4j</exclude>
-                                                    <exclude>ch.qos.logback:logback-classic</exclude>
+                                                    
+                                                    
                                                     <exclude>javax.mail:mail</exclude>
                                                     
                                                     <exclude>org.apache.nifi:nifi-bootstrap</exclude>
@@ -547,7 +547,7 @@ language governing permissions and limitations under the License. -->
                                                     <exclude>org.apache.nifi:nifi-processor-utils</exclude>
                                                     
                                                     <exclude>org.apache.nifi:nifi-properties-loader</exclude>
-                                                    <exclude>org.slf4j:slf4j-api</exclude>
+                                                    
                                                     <exclude>javax.activation:activation</exclude>
                                                     <exclude>org.antlr:antlr-runtime</exclude>
                                                     <exclude>asm:asm</exclude>
@@ -559,7 +559,7 @@ language governing permissions and limitations under the License. -->
                                                     <exclude>com.fasterxml.jackson.core:jackson-databind</exclude>
                                                     <exclude>com.jayway.jsonpath:json-path</exclude>
                                                     <exclude>net.minidev:json-smart</exclude>
-                                                    <exclude>ch.qos.logback:logback-core</exclude>
+                                                    
                                                     <exclude>org.apache.nifi:nifi-security-utils</exclude>
                                                     <exclude>org.apache.nifi:nifi-utils</exclude>
                                                     
@@ -574,14 +574,14 @@ language governing permissions and limitations under the License. -->
                                                 <includes>
                                                     <include>org.bouncycastle:bcprov-jdk15on</include>
                                                     <include>org.apache.commons:commons-lang3</include>
-                                                    <include>org.slf4j:jul-to-slf4j</include>
+                                                    
                                                     <include>ch.qos.logback:logback-classic</include>
                                                     <include>javax.mail:mail</include>
                                                     <include>org.apache.nifi:nifi-api</include>
                                                     <include>org.apache.nifi:nifi-bootstrap</include>
                                                     <include>org.apache.nifi:nifi-expression-language</include>
                                                     <include>org.apache.nifi:nifi-processor-utils</include>
-                                                    <include>org.apache.nifi:nifi-properties</include>
+                                                    
                                                     <include>org.apache.nifi:nifi-properties-loader</include>
                                                     <include>org.slf4j:slf4j-api</include>
                                                     <include>javax.activation:activation</include>
-- 
1.8.3.1

**Comment by olegz** [comment]
Cool, will look and merge in the AM

**Comment by joewitt** [comment]
@olegz i'll take care of merging because I'll need to squash and rebase beforehand.  Will do once +1 received.

**Comment by olegz** [comment]
+1

**Pull request closed without merging**